test/e2e/traceflow_test.go

```diff
@@ -552,6 +553,46 @@ func testTraceflowIntraNode(t *testing.T, data *TestData) {
 			},
 			expectedPhase: v1alpha1.Failed,
 		},
+		{
```

Comment:
it feels to me that new e2e tests may not be strictly required for this, instead we could simply use unit tests?

Reply:
I prefer to use e2e here, since these 2 cases will get reason="timeout" previously.
I'll add reason check for this.

Reply:
Added failed reason check for all related testcases.

Reply:
I also fail unit test is good enough, considering e2e tests already run long.

Reply:
Added UT for this, but I still feel to keep the e2e test cases.